woocommerce/woocommerce-android
# [Simplified login] Happy path for Jetpack activation steps

Closes: #7848 


⚠️ depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2572

### Description
This PR adds implementation for the Jetpack installation/connection steps, it concentrates on the happy path, while the currently implemented error handling is limited on setting the step's status, the proper error handling will be added as part of #7849. 

The WebView's Toolbar will be fixed in #7797.

### Testing instructions
The implementation attempts to handle process-death scenarios, to test it, please enable "Don't keep activities" from the developer options, and rotate your phone/navigate to other apps during the installation steps.

And please make sure the Jetpack Native experiment is used, you can either hardcode the value in `JetpackInstallationExperiment` or set your device as test device in Firebase Console.

##### Case 1: Jetpack installation
1. Use a site that doesn't have Jetpack.
2. Open the app, then sign in using your WordPress.com account.
3. In the site picker, click on "Add a store", then click on "Connect an existing store".
4. Click on "Install Jetpack"
5. Enter site credentials.
6. When the connection step is reached, approve it in the WebView.
7. Click on "Go to store" when the connection is done.
8. Confirm the app loads with the correct site.

##### Case 2: Jetpack connection
1. Use a site that has Jetpack but it's not connected.
2. Open the app, then sign in using your WordPress.com account.
3. In the site picker, click on "Add a store", then click on "Connect an existing store".
4. Click on "Connect Jetpack"
5. Enter site credentials.
6. Continue the steps same as case 1.

### Images/gif
https://user-images.githubusercontent.com/1657201/203312342-46913b53-da59-4c87-bc27-f9b8c05f86d9.mov
